src/content/1.7/development/architecture/domain-exceptions.md

```diff
@@ -0,0 +1,138 @@
+│   ├── CategoryConstraintException.php
+│   ├── CategoryException.php
+│   ├── CategoryNotFoundException.php
+│   ├── FailedToDeleteCategoryException.php
+│   └── MenuThumbnailsLimitException.php
+├── Query
+├── QueryHandler
+├── QueryResult
+└── ValueObject
+```
+
+We can see the `Exception` directory, which contains all category domain exceptions. 
+
+All domain exceptions are extending a base exception class. 
+In case of categories, all category exceptions are extending `CategoryException`, which defines the `Category` domain,
+and `CategoryException` extends `PrestaShop\PrestaShop\Core\Domain\Exception\DomainException`, 
+which defines the PrestaShop's domain.
+
+## Example usage of domain exceptions
+
+Let's analyze one exception from `Category` domain, `CannotEditCategoryException`:
+
+```php
+namespace PrestaShop\PrestaShop\Core\Domain\Category\Exception;
+
+/**
+ * Class CannotEditCategoryException is thrown when editing category fails.
+ */
+class CannotEditCategoryException extends CategoryException
+{
+}
+```
+
+As we can guess from the comment in the code, this exception is supposed to be thrown when editing a category fails.
+Let's see it's real usage in a command handler:
+
+```php
+// src/Adapter/Category/CommandHandler/EditCategoryHandler.php
+
+private function updateCategoryFromCommandData(Category $category, EditCategoryCommand $command)
+{
+    // ... 
+    
+    if (false === $category->update()) {
+        throw new CannotEditCategoryException(
+            sprintf('Failed to edit Category with id "%s".', $category->id)
+        );
+    }
+}
+```
+
+{{% notice tip %}}
+If you are not familiar with command handlers or how CQRS pattern is used in PrestaShop, you can read about it here: [CQRS]({{< ref "1.7/development/architecture/cqrs.md" >}}).
+{{% /notice %}}
+
+We can see in the code above, that if category update fails, it will throw a `CannotEditCategoryException`. 
+The exception then needs to be handled in the upper layers of the code. 
+
+In our example all category domain exceptions are handled in the `CategoryController`:
+
+```php
+// src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php
+
+public function editAction($categoryId, Request $request)
+{
+    try {
+        $editableCategory = $this->getQueryBus()->handle(new GetCategoryForEditing((int) $categoryId));
+    } catch (CategoryException $e) {
+        // Catching all exceptions from category domain
+        // and showing a specific error message for every exception type.
+        $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+    }
+
+    // ...
+}
+```
+
+So now, if we are editing a category and it fails for some reason (e.g. the database is not responding),
+the controller will catch the `CannotEditCategoryException` (which is a child of `CategoryException`) and display a specific error message to the user.
+
+## But there are so many new exception classes. I'm scared!
```

Comment:
```suggestion
## Why are there so many new exception classes.
```
Maybe a bit more professional 😛

Reply:
oh I thought I reached the pinnacle of poetry :sweat_smile: